pronotepy/clients.py

```diff
@@ -354,6 +354,22 @@ def __init__(
     ) -> None:
         super().__init__(pronote_url, username, password, ent, qr_code)
 
+    def report(self, period: dataClasses.Period) -> dataClasses.Report:
```

Comment:
Since you already need the period, I'd move this method directly into the `Period` class. Just like [`.grades`](https://github.com/Skythrew/pronotepy/blob/fc5dc539ed890d50a4ff8421765ff8abde2e5208/pronotepy/dataClasses.py#L480), or `.averages`...